File: EasyPC/EasyPC.Model/CompatibilityIssue.cs
namespace EasyPC.Model;

public class CompatibilityIssue
{
    public string Component { get; set; } = string.Empty;
    public string Issue { get; set; } = string.Empty;
    public string Severity { get; set; } = "Warning"; // "Warning", "Error", "Info"
    public string? Suggestion { get; set; }
}

File: EasyPC/UI/easy_pc_admin/lib/models/compatibility_issue.dart
class CompatibilityIssue {
  final String component;
  final String issue;
  final String severity;
  final String suggestion;

  CompatibilityIssue({
    required this.component,
    required this.issue,
    required this.severity,
    required this.suggestion,
  });

  factory CompatibilityIssue.fromJson(Map<String, dynamic> json) {
    return CompatibilityIssue(
      component: json['component'] as String,
      issue: json['issue'] as String,
      severity: json['severity'] as String,
      suggestion: json['suggestion'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
        'component': component,
        'issue': issue,
        'severity': severity,
        'suggestion': suggestion,
      };
}
